yext/answers-headless-react
# Vertical Links Support

Added Vertical Links Support in VisualSearchbar

- consider vertical links as an option when constructing the list of options to pass to DropdownSection
  - move onSelectOption to onClick as part of Option interface, so link and normal option can defined its own behavior when click on or pressed enter. 
  - updated renderAutocompleteResults to take in css classes and icon to accommodate for different types of option
  - added css styling

- other fixes:
  - Enter event handler in inputDropdown should only run when the target is the input element.
  -  any react update should be avoided when component get unmounted (i.e. browser navigation or async function related). Added guard in useEntityPreview and InputDropdown


J=SLAP-1712
TEST=manual

used mock data for autocompleteResults[].verticalKeys field:
- See that the vertical links are display under each option appropriately.
- See that click/enter on a link navigate to the right vertical search page with the selected query in search bar
- See that keyboard up/down navigation works properly with and without vertical links in InputDropdown
